static/app/components/repositoryEditForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import {IconWarning} from 'app/icons';
+import {t, tct} from 'app/locale';
+import {Repository} from 'app/types';
+import {FieldFromConfig} from 'app/views/settings/components/forms';
+import Form from 'app/views/settings/components/forms/form';
+import {Field} from 'app/views/settings/components/forms/type';
+
+import Alert from './alert';
+
+type Props = Pick<Form['props'], 'onSubmitSuccess' | 'onCancel'> &
+  Partial<Pick<Form['props'], 'onSubmit'>> & {
```

Comment:
@MeredithAnya I think we can clean this up with 
```suggestion
Pick<Form['props'], 'onSubmitSuccess' | 'onCancel' | 'onSubmit'>
```

Reply:
@NisanthanNanthakumar actually looks like we don't use `onSubmit` at all so I just took it out